app/forms/idv/doc_pii_form.rb

```diff
@@ -103,5 +100,12 @@ def dob_error
     def dob_min_age_error
       I18n.t('doc_auth.errors.pii.birth_date_min_age')
     end
+
+    # Future work could include a matching jurisdiction_valid? method since this
+    # does the same thing as the current jurisdiction validation.
+    def state_valid?
```

Comment:
Intriguing.  I'm curious why you wrote a separate method for the state validation, rather than structuring it like the existing validation for the jurisdiction?
```ruby
validates :state, inclusion: { in: Idp::Constants::STATE_AND_TERRITORY_CODES, message: proc { I18n.t('doc_auth.errors.general.no_liveness') }}
```
Both definitely work, I'm mostly curious about your thinking.

Reply:
I did it with the `proc` first and basically thought it wasn't as clear as extracting the method 🤷 I don't feel strongly. I did notice that the jurisdiction validation is identical so I was thinking they could maybe be the same method? Not sure that would help readability in the end though.

Reply:
I'd prefer to keep the `proc` and combine the validations for the two fields, like below:

```suggestion
        validates :jurisdiction,
              :state, inclusion: { in: Idp::Constants::STATE_AND_TERRITORY_CODES,
                                   message: proc {
                                              I18n.t('doc_auth.errors.general.no_liveness')
                                            } }
```

I think it's more concise than having two separate `foo_valid?` methods for them both.

Reply:
That is much more concise. Tried making that change and it seemed like it was doing something unexpected:
- Made that change
- Ran `bundle exec rspec spec/controllers/idv/image_uploads_controller_spec.rb`
- `due to invalid State` fails
    - :error_details=>{:state=>{:state=>true}}, # expected
    - :error_details=>{:state=>{:inclusion=>true}}, # received

Reply:
I looked at that specific spec, and it also failed after the validation for state was changed from a length validation.  I don’t think it’s unexpected that it would fail after another change in validation.  Thoughts?

Reply:
That seems reasonable to me. Made those changes [here](https://github.com/18F/identity-idp/pull/9797/commits/2e4e975c275214e4672b1444b3bb9391f4937175).

 I've been hesitant to make changes simply because I don't know if that inner key is important.